ansible/roles/packages/tasks/redhat.yaml

```diff
@@ -8,13 +8,9 @@
   register: installed_rpms
 
 - name: install common RPMS if already not installed
-  yum:
-    name: "{{ item }}"
-    state: present
-    enablerepo: "{{ 'offline' if offline_mode_enabled else '' }}"
-    disablerepo: "{{ '*' if offline_mode_enabled else '' }}"
-    update_cache: true
-  when: installed_rpms.results | selectattr('name','equalto', item ) | list | count == 0
+  command: yum install {{ '--disablerepo "*" --enablerepo offline --allowerasing' if offline_mode_enabled else '' }} {{ item }}
```

Comment:
Maybe we can try this with `--best` instead of `--allowerasing`?

>dnf --best install tito
Install the latest available version of the package. If the package is already installed it will try to automatically upgrade to the latest version. If the latest version of the package cannot be installed, the installation will fail.

Reply:
When this ran I observed that one package dependency of a package that was updated by our offline repository was the only thing removed.

```sh
[root@ip-10-0-0-237 yum.repos.d]# yum install --disablerepo "*" --enablerepo offline --allowerasing nfs-utils
Last metadata expiration check: 0:01:33 ago on Sat Feb 24 00:16:22 2024.
Dependencies resolved.
======================================================================================================
 Package                     Architecture      Version                     Repository            Size
======================================================================================================
Installing:
 nfs-utils                   x86_64            1:2.5.4-20.el9              offline              425 k
Upgrading:
 keyutils-libs               x86_64            1.6.3-1.el9                 offline               31 k
 libsss_certmap              x86_64            2.9.1-4.el9_3.5             offline               92 k
 libsss_idmap                x86_64            2.9.1-4.el9_3.5             offline               43 k
 libsss_nss_idmap            x86_64            2.9.1-4.el9_3.5             offline               47 k
 libsss_sudo                 x86_64            2.9.1-4.el9_3.5             offline               37 k
 sssd-client                 x86_64            2.9.1-4.el9_3.5             offline              162 k
 sssd-common                 x86_64            2.9.1-4.el9_3.5             offline              1.6 M
Installing dependencies:
 gssproxy                    x86_64            0.8.4-6.el9                 offline              108 k
 keyutils                    x86_64            1.6.3-1.el9                 offline               72 k
 libev                       x86_64            4.33-5.el9                  offline               52 k
 libnfsidmap                 x86_64            1:2.5.4-20.el9              offline               60 k
 libtirpc                    x86_64            1.3.3-2.el9                 offline               92 k
 libverto-libev              x86_64            0.3.2-3.el9                 offline               13 k
 quota                       x86_64            1:4.06-6.el9                offline              190 k
 quota-nls                   noarch            1:4.06-6.el9                offline               78 k
 rpcbind                     x86_64            1.2.6-5.el9                 offline               56 k
 sssd-nfs-idmap              x86_64            2.9.1-4.el9_3.5             offline               42 k
Removing dependent packages:
 sssd-kcm                    x86_64            2.7.3-4.el9_1.3             @anaconda            226 k
 ```

To me this seems pretty minimal. 

I think the strongest arguments here would be:

1. The change to a dependent package of something that was upgraded because our target package required a newer version. 
2. We are in charge of installing these packages and we should have a higher precedence to getting the necessary packages installed.
3. They can update a dependent package if they absolutely need to through our tooling or a satellite.
4. They are using containers OS packages should not be something a customer should be worried about.